platform/mbed_enum_flags.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_SCOPED_ENUM_FLAGS_H
+#define MBED_SCOPED_ENUM_FLAGS_H
```

Comment:
As this is platform addition, might find use in another parts of the codebase, can you send this separately ? It would be useful do describe here using doxygen what it does, how it shall be used, any limitations, etc.

My first question would be why do we need preprocessor magic sugar, if there is a way to avoid  that?

We can then focus in this PR on KVstore changes

Reply:
I'd have no problem doing that. However, I am not quite adequate when it comes to dealing with the Best Way&trade; (or even the Better Way&trade;) of how to implement this.

I started to write this out as a way of adding bitwise operators to scoped enums. After some time of working at this, I realized that as the mbed environment moves to C++11 from the versions that our ancestors chiseled into the 8" stone tablets, more enum types would benefit from this.

All of that to say that if you want this as is, even as a starting point only, I can put it in its own PR. If you want it as some sort of template (beyond what I already managed to crudely implement), you'll have to suggest these in more detail. I will share my thoughts/concerns about implementations in the new PR.

I will work on the PR soon and let the chips fall where they may.